dashboard/api.py

```diff
@@ -367,3 +376,107 @@ def format_courserun_for_dashboard(course_run, status_for_user, certificate=None
         # TODO: here goes the logic to pull the current grade  # pylint: disable=fixme
         pass
     return formatted_run
+
+
+def get_student_enrollments(user, edx_client, course_ids):
+    """
+    Return cached enrollment data or fetch enrollment data first if necessary. Only enrollments with a course key in
+    course_ids are updated on the database, others are left alone.
+
+    Args:
+        user (django.contrib.auth.models.User): A user
+        edx_client (EdxApi): EdX client to retrieve enrollments
+        course_ids (list of str): List of course keys
+    Returns:
+        Enrollments: an Enrollments object from edx_api. This may contain more enrollments than correspond to
+            course_ids if more exist from edX, or it may contain fewer enrollments if they
+            don't exist for the course id.
+    """
+    # Data in database is refreshed after 5 minutes
+    now = datetime.datetime.now(tz=pytz.utc)
+    five_minutes_ago = now - datetime.timedelta(0, 0, 0, 0, 5)
+    enrollments_list = list(models.Enrollment.objects.filter(
```

Comment:
I was worried about race conditions, but I could just wrap the two calls with a transaction instead.